packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -11,11 +11,43 @@ import 'package:flutter/widgets.dart';
 import 'button_style.dart';
 import 'colors.dart';
 import 'constants.dart';
+import 'elevated_button.dart';
+import 'filled_button.dart';
 import 'ink_well.dart';
 import 'material.dart';
 import 'material_state.dart';
+import 'outlined_button.dart';
+import 'text_button.dart';
 import 'theme_data.dart';
 
+/// {@template flutter.material.ButtonStyleButton.iconAlignment}
+/// Determines the alignment of the icon within the widgets such as:
+///   - [TextButton.icon],
+///   - [ElevatedButton.icon],
+///   - [OutlinedButton.icon],
+///   - [FilledButton.icon],
+///   - [FilledButton.tonalIcon].
+///
+/// It is compatible with [TextDirection].
+///   - If textDirection is [TextDirection.ltr] & iconAlignment is [IconAlignment.start],
+///     it aligns the icon at the start of the button.
+///   - If textDirection is [TextDirection.rtl] & iconAlignment is [IconAlignment.start],
+///     it aligns the icon at the end of the button.
+///   - If textDirection is [TextDirection.ltr] & iconAlignment is [IconAlignment.end],
+///     it aligns the icon at the end of the button.
+///   - If textDirection is [TextDirection.rtl] & iconAlignment is [IconAlignment.end],
+///     it aligns the icon at the start of the button.
```

Comment:
I'm not sure if this is the best way to document these conditions. Hans can suggest better structure here.

Reply:
Previous comment is a suggestion for the text.

Reply:
I've applied the suggestion via GitHub. Thank you!